src/org/openlcb/cdi/swing/CdiPanel.java

```diff
@@ -422,11 +424,33 @@ public void runBackup() {
         }
     }
 
+    private String generateFileName() {
+        String fileName = rep.getRemoteNodeAsString();
+        if (!nodeName.isEmpty()) {
+            fileName += "-"+nodeName;
+        }
+        if (rep.getCdiRep() != null && rep.getCdiRep().getIdentification() != null) {
+            // info not present if configuration hasn't been read yet
+            fileName += "-"+rep.getCdiRep().getIdentification().getSoftwareVersion();
+        }
+        fileName += "-"+java.time.LocalDate.now();
+        fileName  += "-"+java.time.LocalTime.now().format( // use default time zone
+                        java.time.format.DateTimeFormatter.ofPattern("HH:mm:ss")
+                    );
+
+        fileName = fileName.replace(" ", "_"); // don't use spaces in file names!
+
+        return "config." + fileName + ".txt";
+    }
+
+
     public void runRestore() {
         // First select a file to save to.
         fci.setDialogTitle("Open configuration restore file");
         fci.rescanCurrentDirectory();
-        fci.setSelectedFile(new File("config." + rep.getRemoteNodeAsString() + ".txt"));
+
+        fci.setSelectedFile(new File(generateFileName()));
```

Comment:
This surely won't work with the date and time now being part of the filename.
Can the file chooser get something like a glob pattern?

Reply:
You're right that this won't end up selecting a file. This goes to the saved directory (if any), but doesn't select a file.

Unfortunately, no, there's no glob capability in the JFileChooser.  Coding one is possible, but it's a non-trivial amount of work.

Another approach might be to present the last filename used by a Save (in the same session) and select that.